prow/repoowners/repoowners_test.go

```diff
@@ -143,26 +171,96 @@ func getTestClient(
 			return nil, nil, err
 		}
 	}
+	cache := make(map[string]cacheEntry)
```

Comment:
Hmm this is a lot of baked-in complexity for a suite of test cases. Can we inject the cache entries directly and have cache entry literals in the test cases?

Reply:
@stevekuznetsov The cache is related to git commit change files and sha, so I used cacheOptions to Dynamically Generate cache entry. It cannot use inject the cache entries directly

Reply:
@stevekuznetsov The cache is related to git commit change files and sha, so I used cacheOptions to Dynamically Generate cache entry. It cannot use inject the cache entries directly

Reply:
Hmm it is a real shame we're using a local `git` and dir instead of a fully mocked out fake here

Reply:
This does have the benefit of testing the git diff code that was added, but I agree that it would be nicer to use a mock here and test the diffing code more directly. I think this is fine for now though.

Reply:
@stevekuznetsov Follow the @cjwagner's advice, Do you think it needs to be modified here? if need, I will fix it by using a mock here.

Reply:
I say leave it for now and readdress after @stevekuznetsov's git client refactor, but I don't feel strongly either way.